ts/state/ducks/conversations.ts

```diff
@@ -2093,6 +2096,40 @@ function retryMessageSend(
   };
 }
 
+export function copy(
+  messageId: string
+): ThunkAction<void, RootStateType, unknown, NoopActionType> {
+  return async dispatch => {
+    const { getAbsoluteAttachmentPath } = window.Signal.Migrations;
+    const message = await getMessageById(messageId);
+    if (!message) {
+      throw new Error(`copy: Message ${messageId} missing!`);
+    }
+
+    const body = (message.get('body') || '').trim();
+    if (body) {
+      clipboard.writeText(body);
+    } else {
+      const attachments = message.get('attachments') || [];
```

Comment:
Sometimes folks send an image with text as well. Seems like this implementation will only copy an image if there's no text. This is a little confusing. Maybe we have the feature be "Copy text" instead?

Reply:
If so, there's a library that does the copying: https://npmjs.com/package/copy-text-to-clipboard

Reply:
@josh-signal We've kinda talked about that above. This implementation is going to let user; always copy the text if there is a text (it'll copy the text no matter what, even though there is an image), else; if there is a single image, copy the image. I would say that's pretty much what people saw in messaging applications.

But, as it's been already so long since this pull request created and i need to use this feature myself too; i'm not going to argue about which label we should give to the copy button, if we want to make sure it's 100% understandable; we can have "Copy Image" button if there is a single image, and "Copy Text" button if there is a text. So user can have both option to copy text or image if there is a message with both text and image.

Also we've already got functions to copy text/image, i think we don't need the library you mentioned.